e2e/internal/e2e/imageverify.go

```diff
@@ -0,0 +1,406 @@
+		if !fileExists(t, file) {
+			t.Fatalf("unexpected failure: %%Setup generated file %v does not exist on host", file)
+		}
+	}
+
+	for _, file := range dfd.Post {
+		if !fileExists(t, filepath.Join(imagePath, file)) {
+			t.Fatalf("unexpected failure: %%Post generated file %v does not exist in container", file)
+		}
+	}
+
+	// Verify any apps
+	for _, app := range dfd.Apps {
+		// %apphelp
+		if app.Help != nil {
+			helpPath := filepath.Join(imagePath, `/scif/apps/`, app.Name, `/scif/runscript.help`)
+			if !fileExists(t, helpPath) {
+				t.Fatalf("unexpected failure in app %v: Script %v does not exist in app", app.Name, helpPath)
+			}
+
+			if err := verifyHelp(t, helpPath, app.Help); err != nil {
+				t.Fatalf("unexpected failure in app %v: app help message: %v", app.Name, err)
+			}
+		}
+
+		// %appenv
+		if app.Env != nil {
+			if err := verifyEnv(t, cmdPath, imagePath, app.Env, []string{"--app", app.Name}); err != nil {
+				t.Fatalf("unexpected failure in app %v: Env in app is incorrect: %v", app.Name, err)
+			}
+		}
+
+		// %applabels
+		if app.Labels != nil {
+			if err := verifyAppLabels(t, imagePath, app.Name, app.Labels); err != nil {
+				t.Fatalf("unexpected failure in app %v: Labels in app are incorrect: %v", app.Name, err)
+			}
+		}
+
+		// %appfiles
+		for _, p := range app.Files {
+			var file string
+			if p.Src == "" {
+				file = p.Src
+			} else {
+				file = p.Dst
+			}
+
+			if !fileExists(t, filepath.Join(imagePath, "/scif/apps/", app.Name, file)) {
+				t.Fatalf("unexpected failure in app %v: File %v does not exist in app", app.Name, file)
+			}
+
+			if err := verifyFile(t, p.Src, filepath.Join(imagePath, "/scif/apps/", app.Name, file)); err != nil {
+				t.Fatalf("unexpected failure in app %v: File %v: %v", app.Name, file, err)
+			}
+		}
+
+		// %appInstall
+		for _, file := range app.Install {
+			if !fileExists(t, filepath.Join(imagePath, "/scif/apps/", app.Name, file)) {
+				t.Fatalf("unexpected failure in app %v: %%Install generated file %v does not exist in container", app.Name, file)
+			}
+		}
+
+		// %appRun
+		if app.Run != nil {
+			scriptPath := filepath.Join(imagePath, "/scif/apps/", app.Name, "scif/runscript")
+			if !fileExists(t, scriptPath) {
+				t.Fatalf("unexpected failure in app %v: Script %v does not exist in app", app.Name, scriptPath)
+			}
+
+			if err := verifyScript(t, scriptPath, app.Run); err != nil {
+				t.Fatalf("unexpected failure in app %v: runscript: %v", app.Name, err)
+			}
+		}
+
+		// %appTest
+		if app.Test != nil {
+			scriptPath := filepath.Join(imagePath, "/scif/apps/", app.Name, "scif/test")
+			if !fileExists(t, scriptPath) {
+				t.Fatalf("unexpected failure in app %v: Script %v does not exist in app", app.Name, scriptPath)
+			}
+
+			if err := verifyScript(t, scriptPath, app.Test); err != nil {
+				t.Fatalf("unexpected failure in app %v: test script: %v", app.Name, err)
+			}
+		}
+	}
+
+}
+
+func fileExists(t *testing.T, path string) bool {
```

Comment:
This seems like it should go into a higher level pkg so it can be reused.

Reply:
What package are you thinking? I figured this `e2e` package is the best place to put helpers like this